### PR TITLE
Make from_hex() safer in rust bindings

### PR DESCRIPTION
**Pull request by jtraglia**
This PR makes it so `from_hex` cannot panic. Previously, it would panic if the string were fewer than 2 bytes or the hex was invalid. This makes the "0x" optional, which is more user friendly IMO; like the Go bindings.

### CODE REVIEW

**Comment by asn-d6** [comment]
This LGTM but let's also get @divagant-martian or @pawanjay176 to check it out.

**Review by pawanjay176 — APPROVED** [review]
LGTM. Just a small suggestion, I'm okay either way.

File: bindings/rust/src/bindings/mod.rs [hunk 1]
```diff
@@ -175,7 +177,10 @@ impl Blob {
     }
 
     pub fn from_hex(hex_str: &str) -> Result<Self, Error> {
-        Self::from_bytes(&hex::decode(&hex_str[2..]).unwrap())
+        let trimmed_str = hex_str.strip_prefix("0x").unwrap_or(hex_str);
```
[the remark this answers is not in the document]
Comment (divagant-martian) [reply bindings/rust/src/bindings/mod.rs:180]: same as @pawanjay176, LGTM and good suggestion!

[the remark this answers is not in the document]
Comment (jtraglia) [reply bindings/rust/src/bindings/mod.rs:180]: Cool, I'll make that change in a few hours; a bit busy right now.

**Comment by asn-d6** [comment]
[121c6bb](https://github.com/ethereum/c-kzg-4844/pull/307/commits/121c6bbc25aea6eb5f0e087a6d500b949f9db50f) LGTM. I will wait for @divagant-martian and @pawanjay176 to take a look as well before merging.

**Review by divagant-martian — APPROVED** [review]
LGTM! nice and clean

**Comment by asn-d6** [comment]
Thanks all!

**Pull request merged**